src/test/java/com/openshift/internal/restclient/DefaultClientFilterIntegrationTest.java

```diff
@@ -0,0 +1,155 @@
+package com.openshift.internal.restclient;
+
+import com.openshift.internal.restclient.model.BuildConfig;
+import com.openshift.internal.restclient.model.project.OpenshiftProjectRequest;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.IResourceFactory;
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IBuildConfig;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IResource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.openshift.internal.restclient.IntegrationTestHelper.*;
+import static com.openshift.restclient.ResourceKind.BUILD_CONFIG;
+import static org.junit.Assert.*;
+
+public class DefaultClientFilterIntegrationTest {
+
+	private static final String VERSION = "v1";
+
+	private static final Logger LOG = LoggerFactory.getLogger(DefaultClientFilterIntegrationTest.class);
+
+	private IClient client;
+
+	private IResourceFactory factory;
+
+	private List<IBuildConfig> buildConfigs = new ArrayList();
+	private IProject project;
+
+	private IntegrationTestHelper helper = new IntegrationTestHelper();
+
+	@BeforeClass
+	public void setup() {
+
+		client = helper.createClientForBasicAuth();
+		factory = new ResourceFactory(client);
+		OpenshiftProjectRequest projectRequest = factory.create(VERSION, ResourceKind.PROJECT_REQUEST);
+		projectRequest.setName(helper.generateNamespace());
+		project = (IProject) client.create(projectRequest);
+
+		buildConfigs.add(createBuildConfigWithLabels(project, "build1", new HashMap<String, String>() {{
+			put("foo", "yes");
+			put("bar", "no");
+			put("baz", "no");
+		}}));
+
+		buildConfigs.add(createBuildConfigWithLabels(project, "build2", new HashMap<String, String>() {{
+			put("foo", "no");
+			put("bar", "yes");
+
+		}}));
+
+		buildConfigs.add(createBuildConfigWithLabels(project, "build3", new HashMap<String, String>() {{
+			put("foo", "yes");
+			put("bar", "yes");
+		}}));
+
+		buildConfigs.add(createBuildConfigWithLabels(project, "build4", new HashMap<>()));
+
+	}
+
+	@AfterClass
+	public void cleanup() {
+		buildConfigs.forEach(bc -> cleanUpResource(client, bc));
```

Comment:
You should be able to just delete the project if you dont want to make individual calls.